src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -189,20 +205,28 @@ public function populate( $thread_id = 0, $order = 'ASC', $args = array() ) {
 			return false;
 		}
 
-		$last_message_index         = count( $this->messages ) - 1;
-		$this->last_message_id      = $this->messages[ $last_message_index ]->id;
-		$this->last_message_date    = $this->messages[ $last_message_index ]->date_sent;
-		$this->last_sender_id       = $this->messages[ $last_message_index ]->sender_id;
-		$this->last_message_subject = $this->messages[ $last_message_index ]->subject;
-		$this->last_message_content = $this->messages[ $last_message_index ]->message;
+		// Messages count.
+		$this->messages_count = self::get_total_thread_message_count( $this->thread_id );
 
-		foreach ( (array) $this->messages as $key => $message ) {
-			$this->sender_ids[ $message->sender_id ] = $message->sender_id;
+		// Sender IDs.
+		$this->sender_ids = self::get_thread_sender_ids( $this->thread_id );
```

Comment:
This introduces a new query, while we used to use `$this->messages` to set the list of `sender_ids` right?
Is this a downsize of introducing messages' thread pagination?

Reply:
Currently, we are requesting ALL messages in a thread. With the change to remove this behavior `$this->messages` won't have all messages so we need another way to get the sender ids.